### DISCUSSION TITLE
Splitting out the Rails Context and React Component to JSON?

### DISCUSSION THREAD

**Discussion by wwahammy** [shakacode/react_on_rails#1530] [Ideas]
So I've been looking at the React on Rails code and I've found that, generally, the code in React on Rails related to sharing data for a Rails context and the information passed to the React components could be reused in other cases. For example, our app has non-React code that needs:

* some context which is included on the page as the view is generated which can be used by the JS on the page
* some component specific information which should be passed to that code as it's first run

I think this is something generally useful and isn't specific to React. In fact, I'd say React on Rails might be a more specialized use of that. Has anyone given thought to splitting that code out into a separate Gem and NPM package and then have React on Rails use those? Or is anyone aware of another implementation of this type of thing that would work in a more general fashion?

**Comment by justin808** [comment]
@wwahammy it would be great to see some sample code better illustrating what you'd like to do. Maybe you can create a very simple rails app to show us?
